scripts/travis/setup-docker-buildx.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+
+set -eux
+
+# update docker
+# for setup buildx https://travis-ci.community/t/installing-docker-19-03/8077/2
+sudo apt update
+
+sudo systemctl stop docker
+sudo apt install -y docker.io
+
+sudo systemctl unmask docker.service
+sudo systemctl unmask docker.socket
+sudo systemctl start docker
+sudo systemctl status docker.socket
+
+docker version
+
+DOCKER_BUILDX_VERSION=v0.4.2
+
+LOCAL_OS=$(uname -s | tr '[A-Z]' '[a-z]')
+
+case $(uname -m) in
+x86_64)
+  LOCAL_ARCH=amd64
+  ;;
+aarch64)
+  LOCAL_ARCH=arm64
+  ;;
+*)
+  echo "unsupported architecture"
+  exit 1
+  ;;
+esac
+
+if [[ ! -f ~/.docker/cli-plugins/docker-buildx ]]; then
+  DOCKER_BUILDX_DOWNLOAD_URL=https://github.com/docker/buildx/releases/download/${DOCKER_BUILDX_VERSION}/buildx-${DOCKER_BUILDX_VERSION}.${LOCAL_OS}-${LOCAL_ARCH}
+  mkdir -p ~/.docker/cli-plugins
+  echo "downloading from ${DOCKER_BUILDX_DOWNLOAD_URL}"
+  curl -sL --output ~/.docker/cli-plugins/docker-buildx "${DOCKER_BUILDX_DOWNLOAD_URL}"
+  chmod a+x ~/.docker/cli-plugins/docker-buildx
+fi
+
+# enable buildx
+export DOCKER_CLI_EXPERIMENTAL=enabled
+
+# checkout buildx available
+docker buildx version
+
+# enabled qemu if needed
+if [[ ! $(docker buildx inspect default | grep Platforms) == *arm64* ]]; then
+  docker run --privileged --rm tonistiigi/binfmt --install all
+fi
+
+# run a local registry for base images push
+# have to do this for multi-arch images
+# multi arch images could not --load
+docker run -d -p 5000:5000 --name registry registry:2
```

Comment:
I am not sure but I'll prefer to push the base image to dockerhub under our organisation instead of starting our own docker-registry in every CI/CD run. That will remove all localhost:5000 too.

Reply:
Agree. I'm actually wondering if we need `docker buildx` at all, given that `buildah` can also generate multi-arch images: 

https://www.youtube.com/watch?v=SYJgkkjqd7s

From what I understand, we don't need qemu or a virtual machine to perform the build, given that we can produce our binaries for the target architectures anyway...

Reply:
@Ashmita152 now in ci, no access to docker.io/jaegertracing for pr.

Reply:
@jpkrohling need to research for this.

Without `qemu`, `RUN apk add ssh` may not be executable.

Reply:
Right, forgot about the `RUN apk add ca` command. Other than that, it would just work, right?

Reply:
@jpkrohling Yes. if just COPY file or config static ENV, CMD ... it could work.

[bazel](https://bazel.build) could do this too. like https://github.com/GoogleContainerTools/distroless did.

Reply:
So, there could be a builder that runs before this and executes `RUN apk add ca`. The platform-specific build would then just copy those files? Sounds like a great solution to me...

Reply:
It could work for base images. But may not for  debug images. 
make devel seems need build platform same sa target platform https://github.com/go-delve/delve/blob/master/_scripts/make.go

Reply:
Even if we don't move to scratch images, we should not start a container registry here (which is the original concern from @Ashmita152).

Reply:
@jpkrohling this question is.

When we use `docker buildx build`, the backend for docker builds named `buildkit` need to pull images from a container registry.

As a multi-arch image, the tag `localhost:5000/baseimg:1.0.0-alpine-3.12` is ref to a manifest list of platform, not a manifest of image layers.
<details>
<summary>example</summary>
<pre><code>
$ DOCKER_CLI_EXPERIMENTAL=enabled docker manifest inspect morlay/baseimg:1.0.0-alpine-3.12
{
   "schemaVersion": 2,
   "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
   "manifests": [
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 737,
         "digest": "sha256:6b22c5fd5b30591e4ce460ee722493e6151e5782804caff18bfee6a21685fbc4",
         "platform": {
            "architecture": "amd64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 737,
         "digest": "sha256:601fe467e2993ecd47ea95b3eb7bb54701cca90da1cfac1af12f59a5b4f76b83",
         "platform": {
            "architecture": "arm64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 737,
         "digest": "sha256:7c6eb953937fa24a2c4d0a34d70d5b9578cdbfd35e42f5e5afdd9fd5b55f7080",
         "platform": {
            "architecture": "s390x",
            "os": "linux"
         }
      }
   ]
}
</code></pre>
</detials>

it could not be storage in docker local. (means we could not use `docker images` to find this, `docker pull morlay/baseimg:1.0.0-alpine-3.12` only pull the image matched local arch)

**so we need push the baseimg with images for muti arch to some container registry**

as @Ashmita152 said, we could push baseimg to `docker.io/jaegertracing`.

But in CI for PR. there have no access to push to `docker.io/jaegertracing`.

This why I start a container registry here to storage images for muti arch

Reply:
Hi @morlay 

> But in CI for PR. there have no access to push to docker.io/jaegertracing.

Can we move the login to docker hub before to provide this access ?

Reply:
it could. 
but be careful. 
it may cause a secure issue in PR

Reply:
>  it may cause a secure issue in PR

Why would it? We should have those as secrets in Travis and they are suppressed from the output in the CI.

> seems Docker username & password not provided for PR

If you set the appropriate env vars in the Travis job for your fork, it should work. You'd have to override the namespace, so that it won't attempt to push to `docker.io/jaegertracing/all-in-one`, but `quay.io/youraccount/all-in-one`.

Reply:
> Why would it? We should have those as secrets in Travis and they are suppressed from the output in the CI.

But i could create some PR and just added some `docker push` to overwrite published images. it not set, the PR may not pass bacause of base image pushing failed.

So I prefer use a local registry
as the said https://github.com/docker/build-push-action#local-registry
 for testing purposes, like PR testings.

Reply:
@jpkrohling 

Or could we create another namespace `docker.io/jaegertracing-dev` or `quay.io/jaegertracing-dev`


`docker.io/jaegertracing` only for master branch
`docker.io/jaegertracing-dev` for testing of PRs or other branches.

Reply:
You'd still have to override the namespace to `jaegertracing-dev`... So, at least for now, you can override to your Dockerhub/Quay namespace.

Reply:
@jpkrohling

in my fork i could override to my Dockerhub/Quay namespace. 
but in PRs of `jaegertracing`, need namespace `jaegertracing-dev` to receive images from PRs.

Without a local registry. base images need to pushing to Dockerhub/Quay in PRs.